README.md

```diff
@@ -1,12 +1,45 @@
-# lambda-template
-Template for creating lambda repositories
-A template for quickly starting a new AWS lambda project.
+# cfn-cr-sc-bucket-policy
 
-## Naming
-Naming conventions:
-* for a vanilla Lambda: `lambda-<context>`
-* for a Cloudformation Transform macro: `cfn-transform-<context>`
-* for a Cloudformation Custom Resource: `cfn-cr-<context>`
+Cloudformation Custom Resource that creates a S3 BucketPolicy. This is intended
+only for use in ServiceCatalog, as it retrieves an ARN that Service Catalog sets,
+`aws:servicecatalog:provisioningPrincipalArn`, to set as a principal on the
+BucketPolicy.
+
+Inventory of source code and supporting files:
+
+- policy_maker - Code for the application's Lambda function.
+- events - Invocation events that you can use to invoke the function.
+- tests - Unit tests for the application code.
+- template.yaml - A template that defines the application's AWS resources.
+
+## Use in a Cloudformation Template
+Create a custom resource in your cloud formation template. Here's an example:
+```yaml
+  S3Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+
+  SCS3BucketPolicy:
+    Type: Custom::SCS3BucketPolicy
+    Properties:
+      ServiceToken: !ImportValue
+        'Fn::Sub': '${AWS::Region}-cfn-cr-sc-bucket-policy-FunctionArn'
+      BucketName: !Ref S3Bucket
+      ExtraPrincipalArns: !Ref S3UserARNs
+      RequireEncryption: True
+```
+
+The creation of the custom resource triggers the lambda. It creates an S3
+BucketPolicy.
+* `ServiceToken` refers to the ARN of the lambda function. You can follow the pattern given; see "Install Lambda into AWS" below for the stack that exports that value.
+* The only required property is `BucketName`, a String.
+* `ExtraPrincipalArns` is one or more valid IAM policy [principals](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html).
+* `RequireEncryption` is a boolean value; it will default to false if left out.
```

Comment:
what would happen if this setting is out of sync with `BucketEncryption` setting?  for example if  BucketEncryption is disabled and this is set to True?  or the other way around?

Reply:
The purpose of `RequireEncryption` is whether those "Deny" statements are included or not. The custom resource makes it possible to match what is currently in the templates. If those templates are wrong and I need to refactor, please let me know.

Reply:
i don't think there's anything wrong here.  i assumed there is a relationship between this param and the 
BucketEncryption param.  i just wanted to understand how they work together or at all.  Maybe I've made a bad assumption and their not related at all?